vortex-array/src/array/constant/canonical.rs

```diff
@@ -26,6 +27,22 @@ impl IntoCanonical for ConstantArray {
             )));
         }
 
+        if let Ok(s) = Utf8Scalar::try_from(self.scalar()) {
+            let const_value = s.value().unwrap();
+            let bytes = const_value.as_bytes();
+            let vec = {
+                let mut v = Vec::with_capacity(self.len());
+                for _ in 0..self.len() {
+                    v.push(bytes)
+                }
+                v
+            };
```

Comment:
this could be `iter::repeat(bytes).take(self.len())` but not sure if it's better